modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -478,64 +520,140 @@ private List<ObjectValidationRule> _getObjectValidationRules(
+			outputObjectValidationRuleSettings,
+			_objectValidationRuleSettingPersistence.findByOVRI_N(
+				objectValidationRule.getObjectValidationRuleId(),
+				ObjectValidationRuleSettingConstants.
+					NAME_OUTPUT_OBJECT_FIELD_ID));
+
+		List<ObjectValidationRuleSetting> updatedObjectValidationRuleSettings =
+			new ArrayList<>();
+
 		for (ObjectValidationRuleSetting objectValidationRuleSetting :
 				objectValidationRuleSettings) {
 
-			_objectValidationRuleSettingLocalService.
-				addObjectValidationRuleSetting(
-					objectValidationRule.getUserId(),
-					objectValidationRule.getObjectValidationRuleId(),
-					objectValidationRuleSetting.getName(),
-					objectValidationRuleSetting.getValue());
+			ObjectValidationRuleSetting
+				serviceBuilderObjectValidationRuleSetting =
+					_objectValidationRuleSettingPersistence.fetchByOVRI_N_V(
+						objectValidationRule.getObjectValidationRuleId(),
+						objectValidationRuleSetting.getName(),
+						objectValidationRuleSetting.getValue());
+
+			if (serviceBuilderObjectValidationRuleSetting != null) {
+				continue;
+			}
+
+			ObjectDefinition objectDefinition =
+				_objectDefinitionPersistence.fetchByPrimaryKey(
+					objectValidationRule.getObjectDefinitionId());
+
+			if (objectDefinition.isApproved()) {
+				ObjectFieldLocalService objectFieldLocalService =
+					_objectFieldLocalServiceSnapshot.get();
+
+				ObjectField objectField =
+					objectFieldLocalService.getObjectField(
+						GetterUtil.getLong(
+							objectValidationRuleSetting.getValue()));
+
+				Table<?> table = objectFieldLocalService.getTable(
+					objectValidationRule.getObjectDefinitionId(),
+					objectField.getName());
+
+				Column<?, ?> column = table.getColumn(
+					objectField.getDBColumnName());
+
+				ObjectEntryLocalService objectEntryLocalService =
+					_objectEntryLocalServiceSnapshot.get();
+
+				long objectEntriesCount =
+					objectEntryLocalService.getObjectEntriesCount(
+						0, objectDefinition, column.isNotNull());
+
+				if (objectEntriesCount > 0) {
+					throw new RequiredObjectFieldException.
+						MustNotUseFilledObjectField();
+				}
+			}
+
+			updatedObjectValidationRuleSettings.add(
+				_objectValidationRuleSettingLocalService.
+					addObjectValidationRuleSetting(
+						objectValidationRule.getUserId(),
+						objectValidationRule.getObjectValidationRuleId(),
+						objectValidationRuleSetting.getName(),
+						objectValidationRuleSetting.getValue()));
 		}
 
-		return newObjectValidationRuleSettings;
+		return updatedObjectValidationRuleSettings;
 	}
 
 	private void _validate(
-			String externalReferenceCode, long objectValidationRuleId,
-			long objectDefinitionId, long companyId, String engine,
-			Map<Locale, String> nameMap, String outputType, String script,
+			boolean active, String externalReferenceCode,
+			long objectValidationRuleId, long objectDefinitionId,
+			long companyId, String engine, Map<Locale, String> nameMap,
+			String outputType, String script,
```

Comment:
@mateussgn please follow the same order from the service.xml, i.e., `boolean active` after `long objectDefinitionId`